zio-kafka-testkit/src/main/scala/zio/kafka/testkit/Kafka.scala

```diff
@@ -105,6 +78,29 @@ object Kafka {
 
   final case class Sasl(value: Kafka) extends AnyVal
 
+  /**
+   * Creates an in-memory Kafka instance with a random port and configured with the provided kafka broker properties.
+   */
+  def withBrokerProps(brokerProps: Ports => Map[String, String]): ZLayer[Any, Throwable, Kafka] =
+    ZLayer.scoped {
+      for {
+        ports <- nextPorts
+        embeddedKafkaConfig = EmbeddedKafkaConfig(
+                                ports.kafkaPort,
+                                ports.zookeeperPort,
+                                brokerProps(ports)
+                              )
+        kafka <- ZIO.acquireRelease(
+                   ZIO
+                     .attempt(EmbeddedKafka.start()(embeddedKafkaConfig))
+                     .map(EmbeddedKafkaService)
+                     .catchSome { case NonFatal(e) =>
+                       throw EmbeddedKafkaStartException("Failed to start embedded Kafka", e)
+                     }
```

Comment:
You shouldn't throw here but return a `ZIO.fail`:
```suggestion
                     .catchSomeZIO { case NonFatal(e) =>
                       ZIO.fail(EmbeddedKafkaStartException("Failed to start embedded Kafka", e))
                     }
```

Reply:
What Jules said, but like this (catchSomeZIO doesn't exist):

```suggestion
                     .catchNonFatalOrDie { e =>
                       ZIO.fail(EmbeddedKafkaStartException("Failed to start embedded Kafka", e))
                     }
```

Reply:
Yeah I wasn't sure it exists 😄